File: algorithms/001.Two_Sum/two_sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        result = [-1, -1]
        num_h = {}
        
        for index in range(len(nums)):
            if (target - nums[index]) in num_h:
                result = [index, num_h[target - nums[index]]]    
            num_h[nums[index]] = index
        
        return result

File: algorithms/001.Two_Sum/two_sum.go
package problem0001

func twoSum(nums []int, target int) []int {
    index := make(map[int]int, len(nums))
    result := []int{-1,-1}
    for i, v := range nums {
        if j, ok := index[target-v]; ok {
            return []int{i, j}
        }
        index[v] = i
    }
    return result
}